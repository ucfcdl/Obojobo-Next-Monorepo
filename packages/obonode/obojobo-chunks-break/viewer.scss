@import '~styles/includes';

.obojobo-draft--chunks--break {
	text-align: center;

	hr {
		border: none;
		border-bottom: 1px solid $color-shadow;
		margin-bottom: 1em;
		margin-top: 1em;
		width: 30%;
	}
}
