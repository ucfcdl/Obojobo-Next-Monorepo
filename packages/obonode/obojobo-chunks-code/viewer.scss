@import '../obojobo-document-engine/src/scss';

.obojobo-draft--chunks--code {
	pre {
		font-size: 1em;
		padding-left: 2em;
		padding-right: 2em;
	}

	.obo-text {
		display: block;
	}
}
