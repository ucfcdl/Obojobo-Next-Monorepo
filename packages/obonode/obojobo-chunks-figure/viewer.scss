@import '~styles/includes';

.obojobo-draft--chunks--figure {
	text-align: center;

	figure {
		margin: 0;
		padding: 0;
		border: 1px solid red;
	}

	.container {
		display: inline-block;
		height: auto;
		position: relative;
		// width: 18em;

		img {
			background: $color-bg;
			border: none;
			display: block;
			width: 100%;
		}

		.img-placeholder {
			border: 1px solid $color-shadow;
		}
	}

	figcaption {
		color: $color-text-minor;
		font-size: 0.9em;
	}

	&.small {
		figure {
			height: 100%;
			margin: 0;
			padding: 0;
			width: 100%;
		}

		.container {
			width: 18em;
		}
	}

	&.medium {
		figure {
			height: 100%;
			margin: 0;
			padding: 0;
			width: 100%;
		}

		.container {
			width: $dimension-column-width;
		}
	}

	&.large {
		figure {
			padding-top: 0;

			img {
				display: block;
			}

			.img-placeholder {
				border-left: none;
				border-right: none;
			}
		}

		.container {
			width: 100%;
		}
	}
}
