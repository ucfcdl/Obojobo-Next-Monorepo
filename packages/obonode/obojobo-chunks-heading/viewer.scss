@import '~styles/includes';

// .obojobo-draft--chunks--heading
// {
// }
