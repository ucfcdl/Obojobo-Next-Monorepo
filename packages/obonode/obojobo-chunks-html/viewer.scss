@import '~styles/includes';

.obojobo-draft--chunks--html {
	display: block;
	margin: 1em auto;
}
