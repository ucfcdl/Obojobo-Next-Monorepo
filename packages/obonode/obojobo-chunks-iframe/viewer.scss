@import '~styles/includes';

.obojobo-draft--chunks--iframe {
	display: block;
	font-size: 0.8em;
	margin: 0 auto;
	max-width: $dimension-editor;
	text-align: left;
	width: 100%;

	iframe {
		height: 20em;
		margin: 0 auto;
		width: 100%;
	}
}
