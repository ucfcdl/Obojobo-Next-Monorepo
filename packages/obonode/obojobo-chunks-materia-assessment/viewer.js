import adapter from 'obojobo-chunks-materia/adapter'
import Common from 'obojobo-document-engine/src/scripts/common'
import ViewerComponent from './viewer-component'

Common.Registry.registerModel('ObojoboDraft.Chunks.MateriaAssessment', {
	adapter: adapter,
	componentClass: ViewerComponent,
	type: 'chunk'
})
