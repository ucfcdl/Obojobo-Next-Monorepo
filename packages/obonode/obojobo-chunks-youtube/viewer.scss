@import '~styles/includes';

.obojobo-draft--chunks--you-tube {
	display: block;
	margin: 1em auto;
	text-align: center;

	iframe {
		height: 25em;
		margin: 0 auto;
		width: 32em;
	}
}
