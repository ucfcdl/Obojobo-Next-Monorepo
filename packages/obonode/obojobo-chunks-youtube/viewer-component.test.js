import React from 'react'
import renderer from 'react-test-renderer'

import YouTube from './viewer-component'
import OboModel from 'obojobo-document-engine/src/scripts/common/models/obo-model'

jest.mock('obojobo-document-engine/src/scripts/common/util/uuid', () => {
	return () => 'mockId'
})

require('./viewer') // used to register this oboModel

describe('YouTube', () => {
	test('YouTube component', () => {
		const model = OboModel.create({
			id: 'id',
			type: 'ObojoboDraft.Chunks.YouTube',
			content: {
				videoId: 'dQw4w9WgXcQ',
				startTime: 4,
				endTime: 20
			}
		})
		const moduleData = {
			focusState: {}
		}

		const component = renderer.create(<YouTube model={model} moduleData={moduleData} />)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})
})
