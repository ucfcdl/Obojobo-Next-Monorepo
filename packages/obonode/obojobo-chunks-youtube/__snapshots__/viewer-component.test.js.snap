// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YouTube YouTube component 1`] = `
<div
  aria-label="YouTube Video"
  className="component"
  data-focussed={false}
  data-id="id"
  data-obo-component={true}
  data-type="ObojoboDraft.Chunks.YouTube"
  id="obo-id"
  role="region"
  tabIndex={-1}
>
  <div
    className="obojobo-draft--chunks--you-tube viewer"
  >
    <div
      className="obojobo-draft--chunks-you-tube-player"
      id="obojobo-draft--chunks-you-tube-player-mockId"
    />
  </div>
</div>
`;
